w11h03/test/pgdp/networking/HTTPTest.java

```diff
@@ -0,0 +1,327 @@
+                            "from_id": 06,
+                            "to_id": 47,
+                            "text": "Don't you remember his name? You know this. Deep down, you know. What was his name?",
+                            "id": 438,
+                            "time": "2022-01-13T10:39:10.000Z"
+                          },
+                          {
+                            "from_id": 47,
+                            "to_id": 06,
+                            "text": "Subject 6. Your name is 6.",
+                            "id": 439,
+                            "time": "2022-01-13T10:40:10.000Z"
+                          },
+                          {
+                            "from_id": 06,
+                            "to_id": 47,
+                            "text": "And what is our purpose?",
+                            "id": 440,
+                            "time": "2022-01-13T10:40:20.000Z"
+                          },
+                          {
+                            "from_id": 47,
+                            "to_id": 06,
+                            "text": "To take them all down.",
+                            "id": 441,
+                            "time": "2022-01-13T10:40:43.000Z"
+                          }
+                        ]
+                        """);
+        var actual = dataHandler.getMessagesWithUser(06, 47, 0);
+        var expected = List.of(
+                new ViewController.Message(ofEpoch(1642070350), "Don't you remember his name? You know this. Deep down, you know. What was his name?", false, 438),
+                new ViewController.Message(ofEpoch(1642070410), "Subject 6. Your name is 6.", true, 439),
+                new ViewController.Message(ofEpoch(1642070420), "And what is our purpose?", false, 440),
+                new ViewController.Message(ofEpoch(1642070443), "To take them all down.", true, 441)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("getMessagesWithUser – unsuccessful")
+    void testGetMessagesFailed() {
+        setField(dataHandler, "id", 47);
+        client.respond(200, """
+                {
+                  "access_token": "RmluZSwgeW91IHdpbi4gTWVzc2FnZSBtZSB0aGUgd29yZCAnZ2FiYnJvJyBvbiBadWxpcC4=",
+                  "token_type": "irrelevant"
+                }""")
+                .respond(422, error);
+        assertNull(dataHandler.getMessagesWithUser(06, 47, 0));
+    }
+
+    @Test
+    @DisplayName("getMessagesWithUser – garbage data")
+    void testGetMessagesGarbage() {
+        setField(dataHandler, "id", 47);
+        client.respond(200, """
+                {
+                  "access_token": "TmV2ZXIgY29udGVudCwgYXJlIHdlPw==",
+                  "token_type": "irrelevant"
+                }""")
+                .respond(500, "This song is new to me, but I am honored to be a part of it." );
+        assertNull(dataHandler.getMessagesWithUser(06, 47, 0));
+    }
```

Comment:
We should check here if this is the correct behavior. I have asked on Zulip https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/Was.20bei.20.60getMessagesWithUser.20.60.20!.3D.20200.20returnen.3F/near/909580.

When we have an answer for this, we should add a comment with the Zulip post.

Reply:
Either `null` or an empty list a valid option: https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/Was.20bei.20.60getMessagesWithUser.20.60.20!.3D.20200.20returnen.3F/near/909580

Therefore, the test should test if the return value is `null` _or_ an empty list.

Reply:
I mean, it says in the exercise, that "From this reply you should build the corresponding List, given that the request was successful.". To me at least, that means that if we can't build it, we have no other option that to return null.

Reply:
> Either `null` or an empty list a valid option: https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/Was.20bei.20.60getMessagesWithUser.20.60.20!.3D.20200.20returnen.3F/near/909580
> 
> Therefore, the test should test if the return value is `null` _or_ an empty list.

Got it. I'll add that case as well.